include/class.dept.php

```diff
@@ -205,6 +200,42 @@ function getAvailableMembers() {
         return $this->getMembers(array('available'=>1));
     }
 
+    function getPrimaryMembers() {
+
+        if (!isset($this->_primary_members)) {
+            $members = clone $this->getMembers();
+            $members->filter(array('dept_id' =>$this->getId()));
+            $members = Staff::nsort($members);
+            $this->_primary_members = $members->all()->asArray();
+        }
+
+        return $this->_primary_members;
+    }
+
+    function getExtendedMembers() {
+
+        if (!isset($this->_exended_members)) {
+            // We need a query set so we can sort the names
+            $members = StaffDeptAccess::objects();
+            $members->filter(array('dept_id' => $this->getId()));
+            $members = Staff::nsort($members, 'staff__');
+            $extended = array();
+            foreach($members->all()->asArray() as $member) {
```

Comment:
You can iterate with `->all()` always. This issue was that it used to return an array and it was changed to return an iterator. #2996 converts it back to an array. Either way, asArray() wouldn't be necessary to iterate